### PR TITLE
Updated API for CUDA Shared Memory support

### PR DESCRIPTION
**Pull request by CoderHam**
- Should SharedMemoryStatus return KIND?

### CODE REVIEW

File: src/core/grpc_service.proto [hunk 1]
```diff
@@ -242,25 +242,82 @@ message SharedMemoryControlRequest
...
+      //@@     The offset from the start of the CUDA shared memory region.
+      //@@     start = offset, end = offset + size;
+      //@@
+      uint64 offset = 2;
+
+      //@@  .. cpp:var:: uint64 byte_size
+      //@@
+      //@@     Size of the CUDA shared memory block, in bytes.
+      //@@
+      uint64 byte_size = 3;
```
Comment (GuanLuo) [inline src/core/grpc_service.proto]: Are the 'offset' and 'byte_size' here the same as those in the outer message? It seems to be a bit confusing without further explanation

File: src/servers/grpc_server.cc [hunk 2]
```diff
@@ -1559,8 +1559,8 @@ SharedMemoryControlHandler::Process(Handler::State* state, bool rpc_ok)
     if (request.has_register_()) {
       err = smb_manager_->Create(
           &smb, request.register_().name(),
-          request.register_().shared_memory_key(), request.register_().offset(),
-          request.register_().byte_size());
+          request.register_().system_shared_memory().shared_memory_key(),
```
Comment (GuanLuo) [inline src/servers/grpc_server.cc]: For now, should add a check on 'system_shared_memory' is set, return unimplemented error if 'cuda_shared_memory' is set instead.

Comment (CoderHam) [reply src/servers/grpc_server.cc]: Thanks! Added a has_system_shared_memory check

File: src/core/grpc_service.proto (hunk 1 above)
Comment (CoderHam) [reply src/core/grpc_service.proto]: **offset**: The offset of the cudaIPC handle from the start of the shared memory region.
**byte_size**: Size of the cudaIPC handle in the shared memory block, in bytes.

The cudaIPC handle is stored in shared memory and the 'key' we used for System Shared memory instead must be a specific shared memory region that contains this cudaIPC handle in the case of the CUDA shared memory

**Comment by deadeyegoodwin** [comment]
@CoderHam Did you run CI for this?

**Comment by CoderHam** [comment]
Was waiting on your final review suggestions, ran it now.

**Pull request merged**